src/renderer/app-update-warning/app-update-warning.ts

```diff
@@ -0,0 +1,120 @@
+/**
```

Comment:
This entire implementation is stateful, when it could be reactive. This is a major root cause for considerable complexity and buggyness in the codebase.